src/core/featuremodel.cpp

```diff
@@ -278,10 +327,48 @@ void FeatureModel::create()
     return;
 
   startEditing();
```

Comment:
No reason to "start editing" if you use the dataprovider to add features after.
However I expect some nasty side-effects of this because when editing directly on the dataprovider, no signals are sent internally about changed features so any caches that might potentially be around will be stale.
Just for your information that we might have to find another approach in the future.

Reply:
What would be the effect of changing the following line to directly work on the layer? Anything else than that `fid` as foreign key would be broken?

Reply:
It affects all solutions where the id is created on saving the feature. AFAIK these are fids on GPKGs. Aren't sequence ids on an online database working like that as well?

I understand that using fids as ids is not very nice anyway, since you only can work with that if you are a "lonely ranger" working with the one and only GPKG. But what's the situation with the online database?

Reply:
> Aren't sequence ids on an online database working like that as well?

There is an option "evaluate default values on server side" in the project for these which works by sending a request to the database on form opening. The difference is that sqlite does not allow to "get the next id from the sequence" (which I am happy to accept as a limitation in sqlite based formats that does not affect our implementation).

Reply:
>  that does not affect our implementation
I totally agree on that. I just had in mind to have a nice solution with the `fid`. I'm not aware of the risks we take with that. 
So we just work on layer level, fine?

Reply:
the risk is that we prevent any usage of "layer.onFeatureAdded" signals (for some features while they are emitted for others) and that we have two different codepaths for different features in general

Reply:
Okay I understand. 

So there will be no fetching after the creation neither. So for what use cases is this used on `save()` here anyway? https://github.com/opengisch/QField/pull/502/files/05fdffed9468ce1db8c0972c8f1039de81d16df8#diff-20da0d6569bb292b15438db05c686d8fL201 
I guess it does that because it needs to receive in the background created values, so I assumed I need it in my use case as well. But maybe I'm wrong anyway.

Reply:
Fetch after creation for the information displayed in the list sounds good if that's not done automatically already now.

Reply:
It's about the parent feature and not about the feature created in the relation editor widget list. And on the fetching there is my situation of confusion. Because I don't have the feature id of the just created feature, I cannot request the feature. That's why I did it with `dataSource` - but still when we don't work with GPKG fids as primaries we probably don't even need it. But then I wonder why it's implemented on `save()`, because I don't want to remove something I'm not sure about that it's still needed...